### PR TITLE
fix(overlay): detach method returns undefined

### LINKED ISSUE #7408: An `undefined` value is returned by `OverlayRef.detach()`
#### Bug, feature request, or proposal:

Bug

#### What is the expected behavior?

```ts
export declare class OverlayRef implements PortalHost {
    /**
     * Detaches an overlay from a portal.
     * @returns Resolves when the overlay has been detached.
     */
    detach(): Promise<any>;
```

#### What is the current behavior?

An `undefined` value is returned by `OverlayRef.detach()`

#### What are the steps to reproduce?

2017-09-29 10_04_41-developer tools - http___localhost_4200_

StackBlitz: https://angular-material2-issue-qqyacc.stackblitz.io/

StackBlitz export: [angular-material2-issue.zip](https://github.com/angular/material2/files/1343620/angular-material2-issue.zip)

The plunker starter btw errors with no modifications:

2017-09-29 10_17_11-developer tools - http___plnkr co_edit_o077b6ueiiigkc0s06dd_p preview

#### What is the use-case or motivation for changing an existing behavior?

Is it a bug?

#### Which versions of Angular, Material, OS, TypeScript, browsers are affected?

`@angular/cdk: 2.0.0-beta.11`

### PR DESCRIPTION
**Pull request by DevVersion**
Currently the JSDoc of the `detach` method describes that the method returns a Promise if the detaching is done. This is not the case and just `undefined` is being returned because every `PortalHost` implementation inside of the CDK is synchronously detaching the view.

In the future it might be useful to change this method back to a `Promise` in favor of animations, but for now it doesn't make sense return a `Promise` because the `attach` method is also synchronous.

Fixes #7408

### CODE REVIEW

File: src/cdk/overlay/overlay-ref.ts [hunk 1]
```diff
@@ -83,11 +83,8 @@ export class OverlayRef implements PortalHost {
     return attachResult;
   }
 
-  /**
-   * Detaches an overlay from a portal.
-   * @returns Resolves when the overlay has been detached.
-   */
-  detach(): Promise<any> {
+  /** Detaches an overlay from a portal. */
+  detach(): void {
```
Comment (crisbeto) [inline src/cdk/overlay/overlay-ref.ts]: Change the `PortalHost` signature as well?

Comment (DevVersion) [reply src/cdk/overlay/overlay-ref.ts]: Not sure about this. I thought that the `PortalHost` can still have the type in case someone wants to implement his own `PortalHost`.

Comment (crisbeto) [reply src/cdk/overlay/overlay-ref.ts]: But since you're removing the return value from the `OverlayRef`, it means that even if they did, they wouldn't be able to use it with our overlays anyway.

Comment (DevVersion) [reply src/cdk/overlay/overlay-ref.ts]: Portals don't have to be used specifically with overlays.

Comment (crisbeto) [reply src/cdk/overlay/overlay-ref.ts]: They don't but that's the primary use case and it's something that we'll end up not supporting. Since the initial goal of the PR was to get rid of the `Promise` signature, consider re-adding the return and setting the return value to `any`?

Comment (DevVersion) [reply src/cdk/overlay/overlay-ref.ts]: That would work too. In that case we would have a very unclear method signature and the return value would be for now always `undefined`. 

Since overlays always seem to use the `DomPortalHost` which seems to return nothing on `detach` I thought we can be clear with this signature here.

I'm fine switching to `any` though.

Comment (DevVersion) [reply src/cdk/overlay/overlay-ref.ts]: Updated as discussed on Slack.
